src/status_im/data_store/contacts.cljs

```diff
@@ -34,6 +35,34 @@
 (defn- get-contact-by-id [public-key realm]
   (core/single (core/get-by-field realm :contact :public-key public-key)))
 
+(defn- get-messages-by-messages-ids
+  [message-ids]
+  (-> @core/account-realm
+      (.objects "messages")
+      (.filtered (str "(" (core/in-query "message-id" message-ids) ")"))))
```

Comment:
this is query is going to be slower than fetching each separate entry using [`.objectForPrimaryKey`](https://realm.io/docs/javascript/latest/api/Realm.html#objectForPrimaryKey)

Reply:
we'll leave that kind of optimization for later this is a mvp and at that point the visible effects of blocking are already applied so perf doesn't matter that much.

Reply:
i agree, but also i would at least consider adding activity indicator because it took 10s on my slower device to block the user which already has sent 1000 messages.

Reply:
What takes 10s is it the realm part or the whole thing. I would think recomputing the message groups of the affected chats is more expensive. Also this is the MVP of the MVP so I'm not really concerned about slow devices at that point

Reply:
most likely realm, because otherwise, each incoming message would have a similar effect. I doubt that those subs have any serious impact here.

It is 1.2-1.5s on galaxy s9 which is not a slow device, that's when removing messages 6 messages with 6 chats and ~2000 messages in total. A realistic test would require 20-30 chats, 15-20k messages in the realm db and removing dozens of messages from different chats. I suppose it might be up to 2.5-3s. 

Well if that's not necessary from UX pov then no problem.